### ISSUE TITLE
Bug: batchnorm not properly transferred to GPU

### ISSUE THREAD

**Issue by RomeoV** [lorenzoh/FastTimm.jl#5]
Not all components get properly transferred to the GPU.
```julia
mv_fn = ToPyTorch("cuda").todevice
effnet = load(models()["timm/efficientnetv2_rw_s"], pretrained=true);
effnet_ = mv_fn.movemodelfn(effnet);
batch = mv_fn.movedatafn(rand(32, 32, 3, 7));

julia> effnet_(batch)
ERROR: PyError ($(Expr(:escape, :(ccall(#= /home/romeo/.julia/packages/PyCall/twYvK/src/
pyfncall.jl:43 =# @pysym(:PyObject_Call), PyPtr, (PyPtr, PyPtr, PyPtr), o, pyargsptr, kw
))))) <class 'RuntimeError'>
RuntimeError('Expected all tensors to be on the same device, but found at least two devi
ces, cuda:0 and cpu! (when checking argument for argument running_mean in method wrapper
__cudnn_batch_norm)')
```
This is fixed by using the pytorch's ` funcexp.replace_all_batch_norm_modules_(model)` as you mentioned [here](https://julialang.zulipchat.com/#narrow/stream/237432-ml-contributers/topic/Training.20PyTorch.20models.20in.20Julia/near/305866791), however, this should probably be automatically done somehow. The questions for me is, why is this the case for batchnorm, and will this also be the case for other modules? 
However, when fixing this, everything works!
```julia
funcexp = pyimport("functorch.experimental")
funcexp.replace_all_batch_norm_modules_(effnet)
julia> effnet_(batch) |> size
# (1792, 7)
```

**Comment by RomeoV** [comment]
Looking [in the pytorch documentation](https://pytorch.org/functorch/stable/batch_norm.html) it seems to have to do with in-place operations, [which are also not supported by Zygote](https://fluxml.ai/Zygote.jl/stable/limitations/#Array-mutation-1).

**Comment by lorenzoh** [comment]
This is a limitation with PyCallChainRules.jl in that it works with functorch modules by using `PyCallChainRules.TorchModuleWrapper` and those in turn have to be `funcexp.replace_all_batch_norm_modules_(effnet)`'d to work. 

FastTimm.jl should do that automatically, I am opening an issue to track this 👍
